modules/generic/src/main/scala/pureconfig/generic/EnumerationConfigReaderBuilder.scala

```diff
@@ -0,0 +1,52 @@
+package pureconfig.generic
+
+import pureconfig.ConfigReader.Result
+import pureconfig.error.NoValidCoproductChoiceFound
+import pureconfig.{ ConfigCursor, ConfigReader }
+import shapeless._
+import shapeless.labelled._
+
+trait EnumerationConfigReaderBuilder[A] {
+  def build(transformName: String => String): ConfigReader[A]
```

Comment:
I might be overlooking something, but doesn't that require us to have two transformation functions: one from type name to config string, and vice versa?

I'd rather have this configurable with a single transformation function. It's the way we're doing with fields in `ProductHint`s, and not all transformation functions are easily invertible. It's also more error-prone because one can easily supply two functions for a converter that are not the inverse of one another.

Reply:
As discussed offline, while I'm still not fond of mixing naming logic with derivation code, I can't see an alternative since we only have the set of valid alternatives at derivation time - the thing we'd need if we wanted to keep using a single function. Let's keep this as it is then.